subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -3182,7 +3249,20 @@ int ztls_getsockopt_ctx(struct tls_context *ctx, int level, int optname,
 			return -1;
 		}
 		return err;
-	} else if (level != SOL_TLS) {
+	}
+
+	if ((level == SOL_SOCKET) && (optname == SO_ERROR) && ctx->error != 0) {
```

Comment:
If `ctx->error == 0`, we then would skip this even if `SO_ERROR` and `SOL_SOCKET` are set. Would this be an issue as then we would call `zsock_getsockopt` few lines below?
Is the error check needed at all?

Reply:
> Would this be an issue as then we would call zsock_getsockopt few lines below?

That is an intended behavior. The idea here is, that if an error on a socket happened due to TLS activity (like for example receiving TLS alert message), the TLS socket layer will set the `error` variable, and that would be returned with `SO_ERROR` option. Otherwise, we forward the `SO_ERROR` option to the TCP's `zsock_getsockopt()`, to check if there's an error on the underlying TCP socket.

Reply:
Ok, that makes sense. It was just not obvious by looking the code, could you add some comment there so that future code reviewer gets the point more easily.

Reply:
Sure - I've added a comment in the code.